src/rulesets/LinearAlgebra/utils.jl

```diff
@@ -27,3 +27,17 @@ function _eyesubx!(X::AbstractMatrix)
 end
 
 _extract_imag(x) = complex(0, imag(x))
+
+# Lower triangle of X - X' overwrite X if possible
+function _sublowertri!(X::AbstractMatrix{T}) where T<:Real
```

Comment:
Is this actually faster than `.-` and `LowerTriangular`?

Reply:
Yes (I just quickly benchmarked it), because it does not compute anything for the upper part of the matrix.